instrumentation/spring-rabbit/src/main/java/brave/spring/rabbit/SpringRabbitTracing.java

```diff
@@ -71,4 +118,39 @@ public SimpleRabbitListenerContainerFactory newSimpleMessageListenerContainerFac
     factory.setAdviceChain(tracingRabbitListenerAdvice);
     return factory;
   }
+
+  /**
+   * Instruments an existing SimpleRabbitListenerContainerFactory
+   */
+  public SimpleRabbitListenerContainerFactory decorateSimpleMessageListenerContainerFactory(
+          SimpleRabbitListenerContainerFactory factory) {
+    Advice[] modifiedAdviceChain = chainListWithTracing(
+            factory.getAdviceChain());
+    factory.setAdviceChain(modifiedAdviceChain);
+    return factory;
+  }
+
+  private Advice[] chainListWithTracing(Advice[] chain) {
+    if (chain == null) {
+      return chain;
```

Comment:
do you want this? or do you want to return a singleton array with the tracingRabbitListenerAdvice?